diff --git a/crates/leptos-mf2/src/text.rs b/crates/leptos-mf2/src/text.rs
index 9604590..1de0d1a 100644
--- a/crates/leptos-mf2/src/text.rs
+++ b/crates/leptos-mf2/src/text.rs
@@ -327,5 +327,5 @@ pub(crate) fn fmt_display<D: Description>(
 ) -> core::fmt::Result {
     #[cfg(not(feature = "ssr"))]
     catalog::track_locale();
-    with_active_text(description, TextUse::Displayed, |text: &str| f.pad(text))
+    with_active_text(description, TextUse::Displayed, |text: &str| f.write_str(text))
 }
